tests/MongoDB.Driver.Tests/AuthenticationTests.cs

```diff
@@ -332,14 +334,19 @@ public void Authentication_succeeds_when_user_has_multiple_credentials_and_mecha
 
         private void AssertAuthenticationSucceeds(MongoClientSettings settings, bool async = false)
         {
-            var client = new MongoClient(settings);
-            if (async)
+            /* If we don't use a DisposableClient, the second run of AuthenticationSucceedsWithMongoDB_X509_mechanism
+             * will fail because the backing Cluster's connections will be associated with a dropped user */
+            using (var client = new DisposableMongoClient(new MongoClient(settings)))
```

Comment:
nit: consider moving this logic to: DriverTestConfiguration

Reply:
Done